### PR TITLE
Patch for PR#20873

### PR DESCRIPTION
**Pull request by ZhenLian**
This is a patch PR to fix a issue raised by https://github.com/grpc/grpc/pull/20873
The reason for the failure is in ALTS integration test, some handshake service, such as portable handshake service, doesn't have a local service account. 
This PR is to remove the check and add some comments on that. 
Please review and let me know if there are some further questions. Thank you so much!

@jiangtaoli2016

### CODE REVIEW

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 1]
```diff
@@ -267,10 +267,8 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
   }
   upb_strview local_service_account =
       grpc_gcp_Identity_service_account(local_identity);
-  if (local_service_account.size == 0) {
-    gpr_log(GPR_ERROR, "Invalid local service account");
-    return TSI_FAILED_PRECONDITION;
-  }
+  // we don't check if local service account is empty here
+  // since portable handshake service doesn't have local identity
```
Comment (jiangtaoli2016) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc]: I would rephrase as 

"We don't check if local service account is empty here because local identity could be empty in certain situations."

**Comment by lidizheng** [comment]
@ZhenLian @jiangtaoli2016 This PR broke our CI test: prod:grpc/core/master/linux/grpc_build_protobuf_at_head.

https://source.cloud.google.com/results/invocations/8a6e9ebb-f7f6-4fb7-ad75-dc8977fe7467/targets
